apps/emqx_resource/src/emqx_resource_worker.erl

```diff
@@ -80,11 +98,13 @@ start_link(Id, Index, Opts) ->
 sync_query(Id, Request, Opts) ->
     PickKey = maps:get(pick_key, Opts, self()),
     Timeout = maps:get(timeout, Opts, infinity),
+    emqx_resource_metrics:matched_inc(Id),
```

Comment:
Matched moved here because otherwise retries would bump the `matched` metric as if they were new requests.